azurerm/resource_arm_iothub_resource_provider.go

```diff
@@ -0,0 +1,261 @@
+package azurerm
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/iothub/mgmt/2017-07-01/devices"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmIotHub() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmIotHubCreateAndUpdate,
+		Read:   resourceArmIotHubRead,
+		Update: resourceArmIotHubCreateAndUpdate,
+		Delete: resourceArmIotHubDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
given we're supporting Import here - could we add a test for it? [Here's an example of how we're doing this for Subnets](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/import_arm_subnet_test.go#L10)

Reply:
This is added :) I'll add more import and resource tests given the time.